### PR TITLE
[skip ci] Fix Broken Link in NYC Taxi Notebook

### PR DESCRIPTION
**Pull request by isVoid**
This PR fixes both broken links for the NYC 2016 and NYC 2017 datasets and reran the notebooks under 22.08.

### CODE REVIEW

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (harrism) [inline notebooks/nyc_taxi_years_correlation.ipynb:58]: <p>Why is there a big slowdown?  Running on a different GPU?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cuspatial/pull/616/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Comment by harrism** [comment]
could probably skip ci for PRs like this?

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (isVoid) [reply notebooks/nyc_taxi_years_correlation.ipynb:58]: @taureandyernv , are you aware how did the notebook run? On what machine, with/without memory pool etc.?

Comment (taureandyernv) [reply notebooks/nyc_taxi_years_correlation.ipynb:58]: you're saving the parquet file as a csv :)

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (taureandyernv) [inline notebooks/nyc_taxi_years_correlation.ipynb:63]: !if [ ! -f \"taxi2017.csv\" ]; then curl 'https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2017-01.parquet' -o taxi2017.csv; else echo \"taxi2017.csv found\"; fi

should be 
!if [ ! -f \"taxi2017.csv\" ]; then curl 'https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2017-01.parquet' -o taxi2017.parquet; else echo \"taxi2017.parquet found\"; fi

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (taureandyernv) [inline notebooks/nyc_taxi_years_correlation.ipynb:83]: `taxi2017 = cudf.read_parquet(\"taxi2017.csv\")`
should be 
`taxi2017 = cudf.read_parquet(\"taxi2017.parquet\")`

**Review by taureandyernv — CHANGES_REQUESTED** [review]
please change the taxi2017.csv to taxi2017.parquet in the noted areas.  Also, that data needs to be downloaded ad its about 4gb. 
 getting the data in the folder, which may be hard for a CI machine without getting ops support, may solve the speed issues

**Comment by taureandyernv** [comment]
Created a PR into your repo @isVoid that has a fully working notebook that also runs fast (esp when data is already downloaded :) ).  Please merge it: https://github.com/isVoid/cuspatial/pull/1

My wall time is ~40 seconds on average, which admittedly is a touch slower than I remember, but maybe by a second or two.

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (harrism) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: @taureandyernv do you have an answer to the question about what hardware it was run on, memory pool etc? We need to make sure this is not a real regression.

Comment (harrism) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: In the future we should always store this information in the notebook.

Comment (taureandyernv) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: We store it in the notebook testing script we built.  So I'm using 2x GV100s for these runs, 3GB to play with on single GPU.  It will take 3 minutes 30 seconds on 2x RTX8000 48GB each (but single GPU)

Comment (taureandyernv) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: <p>acutally...this is about right.  This should take 30ish seconds on a tesla based hardware, 3.5 minutes on quadro.  @isvoid, what slowdown are you talking about?</p>

Comment (harrism) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: If you look at the diff in this PR, you will see the slowdown I am talking about.

The notebook as checked in stores the time taken by the previous run. We should document in the notebook what hardware, RAPIDS and CUDA version, etc. it was run on. Otherwise when people run and they see a slower time they are likely to be confused or surprised.

None of this "this is about right" guesswork. Let's be precise.

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (harrism) [inline notebooks/nyc_taxi_years_correlation.ipynb]: This is the slowdown I'm talking about. More than 2x slower.

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (harrism) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: Here is the slowdown:

old:

```
      "CPU times: user 11.2 s, sys: 6.27 s, total: 17.5 s\n",
      "Wall time: 17.7 s\n"
```

new

```
      "CPU times: user 22.6 s, sys: 14.6 s, total: 37.2 s\n",
      "Wall time: 37.2 s\n"
```

Comment (taureandyernv) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: That is interesting and I've honestly don't recall seeing it run that fast.  However, we can't yet compare these numbers and say there's a regression, but we can strongly suspect it. When we built this notebook for the 0.14 release, [I did my work on RTX8000s (here is the commit with the ~3.5 minute run)](https://github.com/rapidsai/cuspatial/blob/76f0a2c595dc0a0d245a821923b6a14bb8cbcaaf/notebooks/nyc_taxi_years_correlation.ipynb), ran some quick tests on the GV100s to see how my time could be so different, and only reran it on GV100s with this update.  @thomcom used a different machine that was consistently faster with the PIP algorithm.  It was cuSpatial's PIP that helped me realize the depth of the "by algorithm" performance discrepancies that classes of cards in the same arch family could have.  Looking back through the branches, the notebook hasn't actually been reran since its merge and the CI jenkins jobs for each release may have already been deleted.  This may end up being a @thomcom question, as those 17 second numbers were run on his machine, as per https://github.com/rapidsai/cuspatial/commit/6c3d853334a3f038df8f53c3ae74e3f679e22abb and this commit ([which I could only render](https://github.com/rapidsai/cuspatial/blob/bdc8f16fa4d7b0257867c25b5133ee80d0cd6442/notebooks/nyc_taxi_years_correlation.ipynb) ) .  We may need him to rerun the notebook on that system to get any realistic understanding about if there is a regression and where it could have happened.  @thomcom , may you provide your system details for @harrism and, if you still have that system, rerun the notebook on that system on 22.08 nightlies and provide us numbers?

## A bit about the past: 
While I can talk about what I recall observing then (and could be mistaken), the challenge you're facing comes from the stoppage of testing and record keeping for regressions on notebooks right around 0.14 and notebooks testing didn't start back up until after 0.18, which wasn't nearly as thorough until very recently.  Some libraries reestablished forms of testing and record keeping a bit more whole heartedly than others, [but cuGraph remains a shining example of what might help with system spec tracking with their table at the top](https://github.com/rapidsai/cugraph/blob/branch-22.08/notebooks/algorithms/centrality/Centrality.ipynb)).  For now, would it help that current and future cuSpatial notebooks include a version of cuGraph's tables? 

## Thoughts on the future:
I absolutely agree on the precise record keeping, with system specs, which is why it's a built in portion of the outputs with that new beta notebooks testing script (although we currently don't test the cuSpatial notebook).  We're only collecting data, not yet processing or analyzing it.  We're in talks to get it integrated into Ops.  Would you like me to start testing cuSpatial notebooks?  Happy to share more info.

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (taureandyernv) [reply notebooks/nyc_taxi_years_correlation.ipynb]: oh thanks!  I hadn't noticed that at all!

File: notebooks/nyc_taxi_years_correlation.ipynb
Comment (thomcom) [reply notebooks/nyc_taxi_years_correlation.ipynb:60]: Hey @taureandyernv @harrism I'll try to rerun the notebook this afternoon or tomorrow.

**Comment by harrism** [comment]
Since there's still longwinded discussion happening, we have to retarget this to 22.10. We are in code freeze.

**Comment by taureandyernv** [comment]
> Since there's still longwinded discussion happening, we have to retarget this to 22.10. We are in code freeze.

Hmmm, @harrism , i thought the discussion was a side topic of the possibility of a regression, but not dealing with the material updates of the notebook itself.  All necessary changes are complete.  This means that cuspatial won't have a working notebook for 22.08 and there are more eyes on spatial analytics right now...

**Review by harrism — APPROVED** [review]

**Comment by harrism** [comment]
We have rules: PRs must be approved before merging. Also, the labels don't say whether this PR is ready to merge. We should treat opening new PRs during burndown as a "big deal".

I approved this PR, but really @thomcom  should approve it since he is more of a Python codeowner than I am (and since @isVoid can't approve his own PR). I will leave merging up to @isvoid since I don't know if it is ready.

**Comment by isVoid** [comment]
I ran the code on a dgx and branch-22.08 and I have this timing for the cell:
```
CPU times: user 7.28 s, sys: 5.22 s, total: 12.5 s
Wall time: 12.5 s
```
I think the function performance depends a lot on which machine it runs on. Since code freeze starts tomorrow. Let's merge this and follow up on the effects of machine to performance.

**Comment by isVoid** [comment]
@gpucibot merge

**Comment by thomcom** [comment]
I reran the benchmark and got 14.5 seconds.

**Pull request merged**
